test/built-ins/AsyncGeneratorPrototype/return/return-suspendedYield-broken-promise-try-catch.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2022 Chengzhong Wu. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-asyncgenerator-prototype-return
+description: >
+  A broken promise should resume the generator and reject with
+  the exception when the generator's state is suspendedYield.
+info: |
+  AsyncGenerator.prototype.return ( value )
+  ...
+  9. Else if state is suspendedYield, then
+    a. Perform AsyncGeneratorResume(generator, completion).
+  ...
+
+  AsyncGeneratorCompleteStep ( generator, completion, done [ , realm ] )
+  Resumes the steps defined at AsyncGeneratorStart ( generator, generatorBody )
+  ...
+  4. Set the code evaluation state of genContext such that when evaluation is resumed for that execution context the following steps will be performed:
+    ...
+    i. Perform AsyncGeneratorDrainQueue(generator).
+    j. Return undefined.
+
+  AsyncGeneratorDrainQueue ( generator )
+  ...
+  5. Repeat, while done is false,
+    a. Let next be the first element of queue.
+    b. Let completion be Completion(next.[[Completion]]).
+    c. If completion.[[Type]] is return, then
+        i. Set generator.[[AsyncGeneratorState]] to awaiting-return.
+        ii. Perform AsyncGeneratorAwaitReturn(generator).
+        iii. Set done to true.
+  ...
+
+flags: [async]
+features: [async-iteration]
+---*/
+
+var g = async function*() {
+  try {
+    yield;
+    return 'this is never returned';
+  } catch (err) {
+    assert.sameValue(err.message, 'broken promise');
```

Comment:
Since this test is scrutinizing the completion semantics of this function, we should avoid making assertions within the body. Instead, we could store the value in a mutable binding declared in a higher scope, and assert that value in the test's final validation phase.

Reply:
Thank you for your suggestion! Updated :)